datafusion/functions-aggregate/src/count.rs

```diff
@@ -432,6 +432,49 @@ impl GroupsAccumulator for CountGroupsAccumulator {
         Ok(vec![Arc::new(counts) as ArrayRef])
     }
 
+    fn convert_to_state(
+        &self,
+        values: &[ArrayRef],
+        opt_filter: Option<&BooleanArray>,
+    ) -> Result<Vec<ArrayRef>> {
+        let values = &values[0];
+
+        let state_array = match (values.logical_nulls(), opt_filter) {
+            (Some(nulls), None) => {
+                let mut builder = Int64Builder::with_capacity(values.len());
+                nulls
+                    .into_iter()
+                    .for_each(|is_valid| builder.append_value(is_valid as i64));
```

Comment:
I believe `.collect()` should be slightly faster and less verbose than a builder here.

Reply:
even better, we should be able to cast the null array to `int64`

Reply:
FWIW: `into_iter().map().collect::<Int64Array>()` seems to be slower than appending values to the builder :thinking: